common/authorization/interceptor.go

```diff
@@ -82,13 +84,18 @@ func (a *interceptor) Interceptor(
 		// Add auth info to context only if there's some auth info
 		if tlsSubject != nil || len(authHeaders) > 0 {
 			var authHeader string
+			var authExtraHeader string
 			if len(authHeaders) > 0 {
 				authHeader = authHeaders[0]
 			}
+			if len(authExtraHeaders) > 0 {
+				authExtraHeader = authExtraHeaders[0]
+			}
 			authInfo := AuthInfo{
 				AuthToken:     authHeader,
 				TLSSubject:    tlsSubject,
 				TLSConnection: tlsConnection,
+				ExtraData:     authExtraHeader,
```

Comment:
My intent for ExtraData is an open-ended escape hatch for passing additional data if needed. There would have to be a convention between the caller and the claim mapper what goes there and in what format.

I think we should NOT mention id tokens as an intended use of ExtraData even if some people will use it that way, because id tokens in general aren't supposed to be passed for authorization.